Description: Fix FTDI configuration for OpenRD
 The old ft2232 driver was dropped for 0.9.0 and the conversion for
 the OpenRD for the new driver was incorrect, using the incorrect
 FTDI channel and device description. This fixes those up and adds
 the appropriate JTAG speed configuration to get things working again.
Author: Jonathan McDowell <noodles@earth.li>
Forwarded: http://openocd.zylin.com/#/c/4254/
Bug-Debian: https://bugs.debian.org/793214
Last-Update: 2017-10-13
diff --git a/tcl/board/openrd.cfg b/tcl/board/openrd.cfg
index 1051c25..db3cb03 100644
--- a/tcl/board/openrd.cfg
+++ b/tcl/board/openrd.cfg
@@ -3,6 +3,8 @@
 source [find interface/ftdi/openrd.cfg]
 source [find target/feroceon.cfg]
 
+adapter_khz 2000
+
 $_TARGETNAME configure \
 	-work-area-phys 0x10000000 \
 	-work-area-size 65536 \
diff --git a/tcl/interface/ftdi/openrd.cfg b/tcl/interface/ftdi/openrd.cfg
index 8c1a805..9ec5b5f 100644
--- a/tcl/interface/ftdi/openrd.cfg
+++ b/tcl/interface/ftdi/openrd.cfg
@@ -4,15 +4,10 @@
 # http://www.marvell.com/products/embedded_processors/developer/kirkwood/openrd.jsp
 #
 
-echo "WARNING!"
-echo "This file was not tested with real interface, it is based on code in ft2232.c."
-echo "Please report your experience with this file to openocd-devel mailing list,"
-echo "so it could be marked as working or fixed."
-
 interface ftdi
-ftdi_device_desc "OpenRD JTAGKey FT2232D"
+ftdi_device_desc "OpenRD JTAGKey FT2232D B"
 ftdi_vid_pid 0x0403 0x9e90
-ftdi_channel 1
+ftdi_channel 0
 
 ftdi_layout_init 0x0608 0x0f1b
 ftdi_layout_signal nTRST -data 0x0200
